sdk-extensions/incubator/src/main/java/io/opentelemetry/sdk/extension/incubator/fileconfig/AttributesFactory.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.extension.incubator.fileconfig;
+
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.sdk.autoconfigure.internal.SpiHelper;
+import io.opentelemetry.sdk.autoconfigure.spi.ConfigurationException;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.Attributes;
+import io.opentelemetry.semconv.resource.attributes.ResourceAttributes;
+import java.io.Closeable;
+import java.util.List;
+import javax.annotation.Nullable;
+
+final class AttributesFactory
+    implements Factory<Attributes, io.opentelemetry.api.common.Attributes> {
+
+  private static final AttributesFactory INSTANCE = new AttributesFactory();
+
+  private AttributesFactory() {}
+
+  static AttributesFactory getInstance() {
+    return INSTANCE;
+  }
+
+  @Override
+  public io.opentelemetry.api.common.Attributes create(
+      @Nullable Attributes model, SpiHelper spiHelper, List<Closeable> closeables) {
+    if (model == null) {
+      return io.opentelemetry.api.common.Attributes.empty();
+    }
+
+    AttributesBuilder builder = io.opentelemetry.api.common.Attributes.builder();
+
+    String serviceName = model.getServiceName();
+    if (serviceName != null) {
+      builder.put(ResourceAttributes.SERVICE_NAME, serviceName);
+    }
+
+    model
+        .getAdditionalProperties()
+        .forEach(
+            (key, value) -> {
+              if (value == null) {
+                throw new ConfigurationException(
+                    "Error processing attribute with key \"" + key + "\": unexpected null value");
+              }
+              Class<?> valueClass = value.getClass();
+              if (value instanceof String) {
+                builder.put(key, (String) value);
+                return;
+              }
+              if (value instanceof Integer) {
+                builder.put(key, (int) value);
+                return;
+              }
+              if (value instanceof Long) {
+                builder.put(key, (long) value);
+                return;
+              }
+              if (value instanceof Double) {
+                builder.put(key, (double) value);
+                return;
+              }
+              if (value instanceof Float) {
+                builder.put(key, (float) value);
+                return;
+              }
+              if (value instanceof Boolean) {
+                builder.put(key, (boolean) value);
+                return;
+              }
+              if (List.class.isAssignableFrom(valueClass)) {
```

Comment:
Nit: why not just
```suggestion
              if (value instanceof List<?>) {
```

Reply:
Yup, this was a remnant of when I was experimenting with a different strategy. Fixed!